File: ArrayExercise/MaximizeStockProfit/MaximizeStockProfit.py
def findProfit(arr):
    totalprofit=0
    n=len(arr)
    hasStock=False
    buyPrice=0
    for i in range(n-1):
        if(hasStock != True):
            if(arr[i]<arr[i+1]):
               buyPrice= arr[i]
               hasStock = True
            else:
               continue
        else:
            if(arr[i]<arr[i+1]):
                continue
            else:
                totalprofit=totalprofit+ (arr[i]-buyPrice)
                hasStock=False
    if(hasStock):
        if( arr[n-2]<arr[n-1]):
            totalprofit=totalprofit+(arr[n-1]-buyPrice)
        else:
            totalprofit=totalprofit(arr[n-2]-buyPrice)
    else:
        if( arr[n-2]<arr[n-1]):
            hasStock=True
            totalprofit=totalprofit+(arr[n-1]-arr[n-2])
    return totalprofit

arr=[98,178,250,300,40,540,690]
print(findProfit(arr))
           
           

File: ArrayExercise/MaximizeStockProfit/MaximizeStockProfit.java
package ArrayExercise.MaximizeStockProfit;

public class MaximizeStockProfit {

    public static void main(String[] args) {
        int prices[]  = {98,178,250,300,40,540,690};
        // int prices[] ={7,6,4,3,1};
        int boughtPrice=-1,totalProfit=0, i,profit=0;
        boolean hasStock= false;
        for( i=0;i<prices.length-1;i++){
            if(!(hasStock)){
                if(prices[i]>prices[i+1]){
                    continue;
                }else{
                    hasStock=true;
                    boughtPrice=prices[i];
                }
            }else{
                if(prices[i]<prices[i+1]){
                    continue;
                }else{
                    profit= prices[i]-boughtPrice;
                    totalProfit=totalProfit +profit;
                    hasStock=false;
                }
            }
        }
        if(i== prices.length-1 && hasStock){
            profit= prices[i]-boughtPrice;
            totalProfit=totalProfit +profit;
        }

        System.out.println(totalProfit);
    }
}
